include/nshlib/nshlib.h

```diff
@@ -65,6 +65,25 @@
 #  define SCHED_NSH SCHED_FIFO
 #endif
 
+struct nsh_exec_param_s
```

Comment:
why put into public header file, nsh_exec_param_s to nsh_param_s

Reply:
> why put into public header file, nsh_exec_param_s to nsh_param_s

Both "builtin/exec_builtin.c"(`exec_builtin`) and "nshlib/nsh_fileapps.c"(`nsh_fileapp`) need.
Changed to to `nsh_param_s`.

Reply:
but we can put it into https://github.com/apache/nuttx-apps/blob/master/nshlib/nsh.h like exec_builtin and nsh_fileapp.

Reply:
> but we can put it into https://github.com/apache/nuttx-apps/blob/master/nshlib/nsh.h like exec_builtin and nsh_fileapp.

`nsh_fileapp()` is decleared in nshlib, **but** `exec_builtin()` in include/builtin
Maybe we should move it to nshlib first?
- Ddeclaration of nsh_fileapp() and exec_builtin()
```
nshlib/nsh.h:884:int nsh_fileapp(FAR struct nsh_vtbl_s *vtbl, FAR const char *cmd,
include/builtin/builtin.h:79:int exec_builtin(FAR const char *appname, FAR char * const *argv,
```
- Grep "builtin/builtin.h" in nuttx-apps
```
$ grep "\<builtin\/builtin.h\>" . -rn --exclude=tags --binary-files=without-match
./testing/cmocka/cmocka_main.c:36:#include <builtin/builtin.h>
./README.md:63:`exec_builtin()` is defined in the `apps/include/builtin/builtin.h`.
./builtin/Make.dep:42: /home/dramalife/upstream/github/apps/include/builtin/builtin.h \
./builtin/exec_builtin.c:32:#include "builtin/builtin.h"
./nshlib/Make.dep:958: /home/dramalife/upstream/github/apps/include/builtin/builtin.h nsh.h \
./nshlib/nsh_builtin.c:39:#include "builtin/builtin.h"
./include/builtin/builtin.h:2: * apps/include/builtin/builtin.h
```